onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -0,0 +1,571 @@
+      return false;
+  }
+
+  bool op_sum = false;
+  bool op_gemm = false;
+
+  for (auto index : node_indexes) {
+    auto node = graph_viewer.GetNode(index);
+
+
+    //Check if the Operation is Supported by OpenVINO
+    if (!IsOpSupported(node->OpType())) {
+      return false;
+    }
+
+    if (node->OpType() == "Sum") {
+      op_sum = true;
+    }
+
+    if (node->OpType() == "Gemm") {
+      op_gemm = true;
+    }
+
+    //BatchNorm, Conv and Reshape cant take more than 1 input
+    if (node->OpType() == "BatchNormalization" || node->OpType() == "Conv" || node->OpType() == "Reshape") {
+      int count = 0;
+      for (const auto& input : node->InputDefs()) {
+        auto name = input->Name();
+        auto it = initializers.find(name);
+        if (it == initializers.end()) {
+          count++;
+        }
+      }
+      if (count > 1) {
+        return false;
+      }
+    }
+
+    if (dev_id == "MYRIAD" || dev_id == "HDDL") {
+      if (node->OpType() == "Reshape" || node->OpType() == "Unsqueeze" || node->OpType() == "Flatten") {
+        if (input_arrays[0][0] != 1) {
+          return false;
+        }
+      }
+
+      //Disable Inception v1 for MYRIAD and HDDL
+      if (node->OpType() == "Gemm") {
+        int count = 0;
+        for (auto input : node->InputDefs()) {
+          auto name = input->Name();
+          auto it = initializers.find(name);
+          if (it == initializers.end()) {
+            count++;
+          }
+        }
+        if (count > 1) {
+          return false;
+        }
+      }
+    }
+
+
+    //MatMul is only supporetd if it is followed by Add
+    if (node->OpType() == "MatMul") {
+      for (size_t i = 0; i < node->InputDefs().size(); i++) {
+        if (node->InputDefs()[i]->TypeAsProto()->tensor_type().elem_type() != ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_FLOAT) {
+          return false;
+        }
+      }
+
+      auto iter = node->OutputNodesBegin();
+
+      if (iter == node->OutputNodesEnd()) {
+        return false;
+      }
+
+      for (auto it = node->OutputNodesBegin(); it != node->OutputNodesEnd(); ++it) {
+        auto out_node = graph_viewer.GetNode((*it).Index());
+
+        if (out_node->OpType() != "Add") {
+          return false;
+        }
+      }
+
+      bool is_graph_input = false;
+
+      auto graph_inputs = graph_viewer.GetInputs();
+      for (const auto& input : node->InputDefs()) {
+        auto it = find(graph_inputs.begin(), graph_inputs.end(), input);
+        if (it != graph_inputs.end()) {
+          is_graph_input = true;
+        }
+      }
+      if (is_graph_input) {
+        size_t input_dims_size = input_arrays[0].size();
+
+        //Only support 2D Matmul operations
+        for (int i = 0; i < num_inputs; i++) {
+          if (input_arrays[i].size() > 2)
+            return false;
+
+          if (input_dims_size != input_arrays[i].size())
+            return false;
+        }
+      }
+
+      //Disable MNIST for MYRIAD and HDDL
```

Comment:
not sure I understand the comment. 
what aspect of MNIST is not supported?

Reply:
We are disabling MNIST for MYRIAD and HDDL because of an issue in OpenVINO. This issue has not been root caused yet and would need help from OpenVINO team. So for now, we are disabling these so that it can fall back to CPU EP

Reply:
Removed the comment